src/main/java/htmlpublisher/workflow/PublishHTMLStep.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2015 CloudBees Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package htmlpublisher.workflow;
+
+import htmlpublisher.HtmlPublisherTarget;
+import hudson.Extension;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepDescriptorImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepImpl;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+/**
+ * Publishes HTML reports in Workflows.
+ * @author Oleg Nenashev
+ * @since TODO
+ */
+public class PublishHTMLStep extends AbstractStepImpl {
+    
+    /**
+     * The name of the report to display for the build/project, such as "Code Coverage"
+     */
+    private final String reportName;
+
+    /**
+     * The path to the HTML report directory relative to the workspace.
+     */
+    private final String reportDir;
+
+    /**
+     * The file[s] to provide links inside the report directory.
+     */
+    private final String reportFiles;
+
+    /**
+     * If true, archive reports for all successful builds, otherwise only the most recent.
+     */
+    private final boolean keepAll;
+
+    /**
+     * If true, will allow report to be missing and build will not fail on missing report.
+     */
+    private final boolean allowMissing;
```

Comment:
Even if so, you could drop this field; a script wishing for equivalent functionality could just write

```groovy
if (fileExists 'target/reports/x.html') {
    publishHtml reportDir: 'target/reports', …
}
```